### PR TITLE
faster GATKRead.numCigarElements - avoid object allocation for unmodifable list. Speeds up HC runs.

### PR DESCRIPTION
**Pull request by akiezun**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public PileupElement(final GATKRead read,
         Utils.nonNull(read, "read is null");
         Utils.nonNull(currentElement, "currentElement is null");
         Utils.validIndex(baseOffset, read.getLength());
-        Utils.validIndex(currentCigarOffset, read.getCigarElements().size());
+        Utils.validIndex(currentCigarOffset, read.numCigarElements());
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java]: Note: this is the expensive call site

**Comment by lbergelson** [comment]
👍

**Comment by lbergelson** [comment]
@akiezun I gave it a thumbs up, but then I just realized it could probably be improved more.

**Comment by lbergelson** [comment]
SamRecord includes a native `getCigarLength` method which says it may be faster in some situations.

**Comment by lbergelson** [comment]
The implementation should probably call that instead of what it does now.

**Comment by akiezun** [comment]
good point - i'll experiment. https://github.com/broadinstitute/gatk/issues/1797

**Pull request merged**